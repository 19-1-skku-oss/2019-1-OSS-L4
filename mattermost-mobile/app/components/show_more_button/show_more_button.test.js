// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {TouchableOpacity} from 'react-native';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import LinearGradient from 'react-native-linear-gradient';

import ShowMoreButton from './show_more_button';

describe('ShowMoreButton', () => {
    const baseProps = {
        highlight: false,
        onPress: jest.fn(),
        showMore: true,
        theme: Preferences.THEMES.default,
    };

    test('should match, full snapshot', () => {
        const wrapper = shallow(
            <ShowMoreButton {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match, button snapshot', () => {
        const wrapper = shallow(
            <ShowMoreButton {...baseProps}/>
        );

        expect(wrapper.instance().renderButton(true, {button: {}, sign: {}, text: {}})).toMatchSnapshot();
        expect(wrapper.instance().renderButton(false, {button: {}, sign: {}, text: {}})).toMatchSnapshot();
    });

    test('should LinearGradient exists', () => {
        const wrapper = shallow(
            <ShowMoreButton {...baseProps}/>
        );

        expect(wrapper.find(LinearGradient).exists()).toBe(true);
        wrapper.setProps({showMore: false});
        expect(wrapper.find(LinearGradient).exists()).toBe(false);
    });

    test('should call props.onPress on press of TouchableOpacity', () => {
        const onPress = jest.fn();
        const wrapper = shallow(
            <ShowMoreButton
                {...baseProps}
                onPress={onPress}
            />
        );

        wrapper.find(TouchableOpacity).props().onPress();
        expect(onPress).toHaveBeenCalledTimes(1);
    });
});
