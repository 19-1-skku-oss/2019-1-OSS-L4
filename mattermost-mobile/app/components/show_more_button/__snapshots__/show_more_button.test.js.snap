// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton should match, button snapshot 1`] = `
<View
  style={Object {}}
>
  <Text
    style={Object {}}
  >
    +
  </Text>
  <FormattedText
    defaultMessage="Show More"
    id="post_info.message.show_more"
    style={Object {}}
  />
</View>
`;

exports[`ShowMoreButton should match, button snapshot 2`] = `
<View
  style={Object {}}
>
  <Text
    style={Object {}}
  >
    -
  </Text>
  <FormattedText
    defaultMessage="Show Less"
    id="post_info.message.show_less"
    style={Object {}}
  />
</View>
`;

exports[`ShowMoreButton should match, full snapshot 1`] = `
<View>
  <LinearGradient
    colors={
      Array [
        "rgba(255,255,255,0)",
        "rgba(255,255,255,0.75)",
        "#ffffff",
      ]
    }
    end={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={
      Array [
        0,
        0.7,
        1,
      ]
    }
    start={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 50,
        "position": "absolute",
        "top": -50,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "position": "relative",
        "top": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "flex": 1,
          "height": 1,
          "marginRight": 10,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(61,60,64,0.2)",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 37,
          "paddingHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2389d7",
              "fontSize": 16,
              "fontWeight": "600",
              "marginRight": 8,
            }
          }
        >
          +
        </Text>
        <FormattedText
          defaultMessage="Show More"
          id="post_info.message.show_more"
          style={
            Object {
              "color": "#2389d7",
              "fontSize": 13,
              "fontWeight": "600",
            }
          }
        />
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "flex": 1,
          "height": 1,
          "marginLeft": 10,
        }
      }
    />
  </View>
</View>
`;
